pkg/vpcmodel/nodesExplainability.go

```diff
@@ -131,6 +133,20 @@ func (c *VPCConfig) GetNodesWithinAddress(ipAddress string) (networkInterfaceNod
 	return networkInterfaceNodes, nil
 }
 
+// getNodesOfVsi is given a string name or UID of VSI, and
+// returns the list of all nodes with this vsi name or UID
+func (c *VPCConfig) GetNodesOfVsi(vsi string) (vsiNodes []Node, err error) {
+	for _, nodeSet := range c.NodeSets {
+		if nodeSet.Kind() == ResourceTypeVSI && (nodeSet.Name() == vsi || nodeSet.UID() == vsi) {
```

Comment:
I think we can avoid checking `nodeSet.Kind() == ResourceTypeVSI` (the VSI is ibm-specific concept), and just return the NodeSet object matching the input name/UID.. 
@ShiriMoran  please confirm

Reply:
@adisos `NodeSet` can also be e.g. `subnet`; if the user entered a `subnet` name (not very likely but possible) we may get unexpectable results.
Perhaps we can move the function to `ibmvpc`?

Reply:
But what if the given name matches exactly one NodeSets which is a subnet and not a VSI?

Reply:
Maybe add a "todo" comment to handle this later.. we have another open issue (#380)  for adding abstract vpc and subnet, which could be used here..

Reply:
@olasaadi99 please add todo